File: ios/Detonator/Sources/Detonator/Request/VideoPauseRequest.swift
class VideoPauseRequest: Request {
    override public func run() {
        let edge = getComponentEdge()!
        
        let element = edge.children[0].element as! VideoElement
        
        do {
            try element.pause()
            
            end()
        } catch {
            self.error(error: error)
        }
    }
}

File: android/Detonator/src/main/java/com/iconshot/detonator/request/VideoPauseRequest.java
package com.iconshot.detonator.request;

import com.iconshot.detonator.Detonator;
import com.iconshot.detonator.element.videoelement.VideoElement;
import com.iconshot.detonator.renderer.Edge;

public class VideoPauseRequest extends Request {
    public VideoPauseRequest(Detonator detonator, IncomingRequest incomingRequest) {
        super(detonator, incomingRequest);
    }

    public void run() {
        Edge edge = getComponentEdge();

        VideoElement element = (VideoElement) edge.children.get(0).element;

        try {
            element.pause();

            end();
        } catch (Exception e) {
            error(e);
        }
    }
}
